x-pack/plugins/ml/common/util/runtime_field_utils.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPopulatedObject } from './object_utils';
+import {
+  RUNTIME_FIELD_TYPES,
+  RuntimeType,
+} from '../../../../../src/plugins/data/common/index_patterns';
+import type { RuntimeField, RuntimeMappings } from '../types/fields';
+
+export function isRuntimeField(arg: unknown): arg is RuntimeField {
```

Comment:
Looking into this typeguard implementation I think it's worth writing some unit tests

Reply:
This is actually the same implementation in Transform. Considering we have a few other PRs to switch to estypes, I'll leave this here for now so we can consolidate later.